app/services/setting_registry.rb

```diff
@@ -52,9 +65,26 @@ def []=(name, value)
     definition = find(name)
     raise ::Foreman::Exception.new(N_("Setting definition for '%s' not found, can not set"), name) unless definition
     db_record = _find_or_new_db_record(name)
-    db_record.value = value
-    db_record.save!
-    definition.value = db_record.value
+    db_record.update!(value: value)
+  end
+
+  def set_user_value(name, value)
+    definition = find(name)
+    raise ActiveRecord::RecordNotFound.new(_("Setting definition for '%s' not found, can not set") % name, Setting, name) unless definition
```

Comment:
`N_()` instead of `_()`

Reply:
This is not foreman exception, but ActiveRecord, so we need to translate here.